src/Symfony/Component/Security/Http/EventListener/IsCsrfTokenValidAttributeListener.php

```diff
@@ -49,4 +56,18 @@ public static function getSubscribedEvents(): array
     {
         return [KernelEvents::CONTROLLER_ARGUMENTS => ['onKernelControllerArguments', 25]];
     }
+
+    private function getTokenId(string|Expression $id, Request $request, array $arguments): string
+    {
+        if (!$id instanceof Expression) {
+            return $id;
+        }
+
+        $this->expressionLanguage ??= new ExpressionLanguage();
+
+        return (string)$this->expressionLanguage->evaluate($id, [
```

Comment:
why the cast to string?

```suggestion
        return (string) $this->expressionLanguage->evaluate($id, [
```

Reply:
because `\Symfony\Component\ExpressionLanguage\ExpressionLanguage::evaluate` returns `mixed`, so to ensure we always get a string to be passed as `$id` constructor parameter of `\Symfony\Component\Security\Csrf\CsrfToken` that expects a string